test/views/pages/home/navigation_home_test.dart

```diff
@@ -1,54 +1,29 @@
 import "package:flutter/material.dart";
 import "package:flutter_test/flutter_test.dart";
 import "package:hooks_riverpod/hooks_riverpod.dart";
-import "package:proxima/viewmodels/home_view_model.dart";
 import "package:proxima/views/home_content/feed/post_feed.dart";
 import "package:proxima/views/navigation/bottom_navigation_bar/navigation_bar_routes.dart";
 import "package:proxima/views/navigation/bottom_navigation_bar/navigation_bottom_bar.dart";
 import "package:proxima/views/navigation/leading_back_button/leading_back_button.dart";
-import "package:proxima/views/navigation/routes.dart";
 import "package:proxima/views/pages/home/home_page.dart";
 import "package:proxima/views/pages/new_post/new_post_page.dart";
-import "../../viewmodels/mock_home_view_model.dart";
-import "../utils/mock_data/mock_posts.dart";
+
+import "../../../mocks/providers/provider_homepage.dart";
 
 void main() {
-  late MaterialApp homePageApp;
-  late ProviderScope emptyMockedPage;
-  late ProviderScope nonEmptyMockedPage;
+  late ProviderScope emptyHomePageWidget;
+  late ProviderScope nonEmptyHomePageWidget;
 
   setUp(() async {
-    homePageApp = MaterialApp(
-      onGenerateRoute: generateRoute,
-      initialRoute: Routes.home.name,
-    );
-
-    emptyMockedPage = ProviderScope(
-      overrides: [
-        postOverviewProvider.overrideWith(
-          () => MockHomeViewModel(),
-        ),
-      ],
-      child: homePageApp,
-    );
-
-    nonEmptyMockedPage = ProviderScope(
-      overrides: [
-        postOverviewProvider.overrideWith(
-          () => MockHomeViewModel(
-            build: () async => testPosts,
-          ),
-        ),
-      ],
-      child: homePageApp,
-    );
+    emptyHomePageWidget = emptyHomePageProvider;
+    nonEmptyHomePageWidget = nonEmptyHomePageProvider;
   });
```

Comment:
By humble curiosity, why did you choose to do this this way and not simply replacing `emptyHomePageWidget` and `nonEmptyPageWidget` by `emptyHomePageProvider` and `nonEmptyHomePageProvider`? Similarly, why not simply doing
```
ProviderScope emptyHomePageWidget = emptyHomePageProvider;
ProviderScope nonEmptyHomePageWidget = nonEmptyHomePageProvider;
```

Similar question for most of tests in the `pages` package (when the provider does not depend on some parameter).

Reply:
You mean defining it in the setup instead of doing it once at the beginning ?

Reply:
In most of the tests in the `pages` package, you do something like:

```dart
late ProviderScope xWidget;

setUp(() async {
   xWidget = xProvider;
});

testWidget("", (tester) async {
     await tester.pumpWidget(xWidget);
});
```

Why not replacing this by:
```dart
testWidget("", (tester) async {
     await tester.pumpWidget(xProvider);
});
```

Or, if you want to keep the modularity of an external variable (which I personally don't think is necessary):
```dart
ProviderScope xWidget = xProvider;

testWidget("", (tester) async {
     await tester.pumpWidget(xWidget);
});
```

Reply:
I see what you mean.
To be fair, it was the way most tests were written (all except for the circular value it seems), by putting the provider in the setup, so I kept this convention.

Reply:
Wasn't this because creating a provider was lot of code? Now that they are already defined somewhere else, is it really necessary to keep those variables?

Reply:
fixed in 99810edc31106c3d02c2945ae2fe730ba6a58bc6.
I didn't replace the providers that took a parameter, for abstraction purpose.